src/commands/init/index.ts

```diff
@@ -47,26 +66,85 @@ export default class Init extends Command {
+            description: `Package to initialize`,
+            required: false,
+            default: '',
+        }),
+        specifier: Flags.string({
+            char: 's',
+            aliases: ['specifier'],
+            description: `Version specifier for the specified package`,
+            required: false,
+            default: '',
+            dependsOn: ['package'],
+        }),
     };
 
     async run(): Promise<void> {
-        const { flags } = await this.parse(Init);
+        const { flags }: { flags: InitFlags } = await this.parse(Init);
         this.log(`Initializing Velocitas packages ...`);
-        const projectConfig = this.initializeOrReadProject();
+        const projectConfig = this._initializeOrReadProject();
 
-        const appManifestData = AppManifest.read(projectConfig.getVariableMappings().get(APP_MANIFEST_PATH_VARIABLE));
 
-        await this.ensurePackagesAreDownloaded(projectConfig, flags.force, flags.verbose);
+        if (flags.package) {
+            await this._handleSinglePackageInit(projectConfig, flags);
+        } else {
+            await this._handleCompletePackageInit(projectConfig, flags);
+        }
+
+        this._createProjectLockFile(projectConfig, flags.verbose);
+    }
+
+    private async _handleCompletePackageInit(projectConfig: ProjectConfig, flags: InitFlags) {
+        await this._ensurePackagesAreDownloaded(projectConfig.getPackages(), flags);
         projectConfig.validateUsedComponents();
 
         if (!flags['no-hooks']) {
-            await this.runPostInitHooks(projectConfig, appManifestData, flags.verbose);
+            await this._runPostInitHooks(projectConfig.getComponents(), projectConfig, flags.verbose);
         }
+    }
 
-        this.createProjectLockFile(projectConfig, flags.verbose);
+    private async _handleSinglePackageInit(projectConfig: ProjectConfig, flags: InitFlags) {
+        const requestedPackageConfig = new PackageConfig({ repo: flags.package, version: flags.specifier });
+        await this._resolveVersion(requestedPackageConfig, flags.verbose);
+
+        const packageUpdated = projectConfig.updatePackageConfig(requestedPackageConfig);
+        if (packageUpdated) {
+            this.log(`... Updating '${requestedPackageConfig.getPackageName()}' to version '${requestedPackageConfig.version}' in .velocitas.json`);
+        } else {
+            projectConfig.addPackageConfig(requestedPackageConfig);
+            this.log(`... Package '${requestedPackageConfig.getPackageName()}:${requestedPackageConfig.version}' added to .velocitas.json`);
+        }
+
+        await this._ensurePackagesAreDownloaded([requestedPackageConfig], flags);
+        this._finalizeSinglePackageInit(requestedPackageConfig, projectConfig);
+
+        if (!flags['no-hooks']) {
+            let components = projectConfig.getComponentsForPackageConfig(requestedPackageConfig);
+            await this._runPostInitHooks(components, projectConfig, flags.verbose);
+        }
```

Comment:
if (!flags['no-hooks']) {
            await this._runPostInitHooks(projectConfig.getComponentsForPackageConfig(requestedPackageConfig), projectConfig, flags.verbose);
        }